crdt.go

```diff
@@ -311,23 +311,69 @@ func (store *Datastore) handleNext() {
 			continue
 		}
 
-		c, err := cid.Cast(data)
+		bCastHeads, err := store.decodeBroadcast(data)
 		if err != nil {
 			store.logger.Error(err)
 			continue
 		}
 
-		//go func(c cid.Cid) {
-		err = store.handleBlock(c)
+		// For each head, we process it.
+		for _, head := range bCastHeads {
+			//go func(c cid.Cid) {
+			err = store.handleBlock(head)
+			if err != nil {
+				store.logger.Error(err)
+			}
+			//}(c)
+			store.seenHeadsMux.Lock()
+			store.seenHeads[head] = struct{}{}
+			store.seenHeadsMux.Unlock()
+		}
+
+		// TODO: We should store trusted-peer signatures associated to
+		// each head in a timecache. When we broadcast, attach the
+		// signatures (along with our own) to the broadcast.
+		// Other peers can use the signatures to verify that the
+		// received CIDs have been issued by a trusted peer.
+	}
+}
+
+func (store *Datastore) decodeBroadcast(data []byte) ([]cid.Cid, error) {
+	// Make a list of heads we received
+	bcastData := pb.CRDTBroadcast{}
+	err := proto.Unmarshal(data, &bcastData)
+	if err != nil {
```

Comment:
I assume you _did not_ ever support CIDv1? This code will fail to parse that. If you did, I'd actually swap these:

1. Try parsing as a CID first. CIDs must start with 0x12 or 0x01 and only 0x12 is a valid protobuf.
2. Reserve (and forbid) the field with tag 2 from the protobuf. That way, no protobuf will parse as a valid CID.

Reply:
Cidv1 was never supported here. DAG nodes are dag-pb and go-merkledag Cid() method for those nodes always used v0. I would leave things as they are then.